.github/workflows/macos.yaml

```diff
@@ -0,0 +1,38 @@
+name: Test build on MacOS
+
+on:
+  push:
+    branches: [ "master", "auto" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: macos-latest
```

Comment:
Please move this build test to github/workflows/default.yaml matrix (or prove that it should not be integrated with the existing build test in default.yaml despite significant duplication of non-trivial code/logic).